### PR TITLE
Feature/policy activation with lock rebuild

### PR DESCRIPTION
**Pull request by meirhalachmi**

### CODE REVIEW

File: src/corona_hakab_model/update_matrix.py [hunk 1]
```diff
@@ -99,8 +99,9 @@ def reset_policies_by_connection_type(self, connection_type, agents_ids_to_reset
                 self.manager.agents_connections_coeffs[i, connection_type] = 1
 
         # letting all conditioned policies acting upon this connection type know they are canceled
-        for conditioned_policy in self.consts.connection_type_to_conditioned_policy[connection_type]:
-            conditioned_policy.active = False
+        if connection_type in self.consts.connection_type_to_conditioned_policy:
```
Comment (meirhalachmi) [inline src/corona_hakab_model/update_matrix.py:102]: preventing a bug that shouldn't happen, but whatever :)

File: src/corona_hakab_model/update_matrix.py [hunk 2]
```diff
@@ -118,26 +119,27 @@ def apply_policy_on_circles(self, policy: Policy, circles: Iterable[SocialCircle
...
+        if conditioned_policy.reset_current_limitations:
+            self.reset_policies_by_connection_type(con_type)
+        affected_circles = self.apply_policy_on_circles(conditioned_policy.policy, circles)
+        conditioned_policy.active = True
+        # adding the message
+        self.manager.policy_manager.add_message_to_manager(conditioned_policy.message)
+        return affected_circles
+
+    @staticmethod
+    def should_apply_policy(conditioned_policy, manager):
```
Comment (tal-nc) [inline src/corona_hakab_model/update_matrix.py]: should be inside conditioned policy

**Review by tal-nc — APPROVED** [review]
had 1 small comment, but looks great overall

File: src/corona_hakab_model/update_matrix.py (hunk 2 above)
Comment (meirhalachmi) [reply src/corona_hakab_model/update_matrix.py]: Great idea

Comment (meirhalachmi) [reply src/corona_hakab_model/update_matrix.py]: did it

**Pull request merged**